packages/python-runner/src/PythonRunner.ts

```diff
@@ -18,35 +18,49 @@ import { Kernel } from '@jupyterlab/services';
 import { CodeEditor } from '@jupyterlab/codeeditor';
 import { Dialog, showDialog } from '@jupyterlab/apputils';
 
+const RUN_BUTTON_CLASS = 'elyra-PythonEditor-Run';
+
 /**
  * Class: An enhanced Python Script Editor that enables developing and running the script
  */
 export class PythonRunner {
+  id: string;
   kernel: Kernel.IKernel;
   model: CodeEditor.IModel;
   kernelSettings: Kernel.IOptions;
+  startingKernel: boolean;
 
   /**
    * Construct a new runner.
    */
-  constructor(model: CodeEditor.IModel) {
+  constructor(model: CodeEditor.IModel, id: string) {
+    this.id = id;
     this.kernel = null;
     this.model = model;
+    this.startingKernel = false;
   }
 
+  startKernelLock(lock: boolean): void {
+    this.startingKernel = lock;
+    (document.querySelector(
+      '#' + this.id + ' .' + RUN_BUTTON_CLASS
+    ) as HTMLInputElement).disabled = lock;
+  }
   /**
    * Function: Starts a python kernel and executes code from file editor.
    */
   runPython = async (
     kernelSettings: Kernel.IOptions,
     handleKernelMsg: Function
   ): Promise<any> => {
-    if (!this.kernel) {
+    if (!this.kernel && !this.startingKernel) {
       const model = this.model;
       const code: string = model.value.text;
 
       try {
+        this.startKernelLock(true);
         this.kernel = await this.startKernel(kernelSettings);
+        this.startKernelLock(false);
```

Comment:
I think it might be better to keep the lock enabled for the duration of the run or until the 'stop' button is selected.  I don't really see any reason to enable the run button when the kernel is already running and its enablement just makes it easier to disrupt things.

Unrelated but thought I'd raise here...
I would recommend using unique text (however slight it is) in error messages so that the location of where the error occurred can be identified since what we're usually provided is a description or screenshot at best.  A couple lines below we have duplicate messages:  'Could not start kernel environment to execute script.'  I suggest changing one to differentiate it from the other.

Reply:
the reason the lock is only implemented here is because my previous PR added support for interrupting runs by starting a new run. The edge case this is addressing is just more common with that new feature.